File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/core/JavaScriptExceptionWrapper.kt
package ca.logaritm.dezel.core

/**
 * @class JavaScriptExceptionWrapper
 * @since 0.1.0
 */
internal class JavaScriptExceptionWrapper(handler: JavaScriptExceptionHandler) {

	/**
	 * @property handler
	 * @since 0.7.0
	 * @hidden
	 */
	private val handler: JavaScriptExceptionHandler

	/**
	 * @constructor
	 * @since 0.1.0
	 */
	init {
		this.handler = handler
	}

	/**
	 * @method execute
	 * @since 0.1.0
	 */
	public fun execute(error: JavaScriptValue) {
		this.handler(error)
	}
}

File: platforms/ios/Dezel/Dezel/Core/JavaScriptExceptionWrapper.swift
/**
 * @class JavaScriptExceptionWrapper
 * @since 0.1.0
 * @hidden
 */
internal final class JavaScriptExceptionWrapper: NSObject {

	//--------------------------------------------------------------------------
	// MARK: Properties
	//--------------------------------------------------------------------------

	/**
	 * @property context
	 * @since 0.1.0
	 * @hidden
	 */
	internal var context: JavaScriptContext

	/**
	 * @property callback
	 * @since 0.1.0
	 * @hidden
	 */
	internal var callback: JavaScriptExceptionHandler

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * @constructor
	 * @since 0.1.0
	 * @hidden
	 */
	internal init(context: JavaScriptContext, callback: @escaping JavaScriptExceptionHandler) {
		self.context = context
		self.callback = callback
	}
}
